server.lua

```diff
@@ -399,6 +401,76 @@ function GetGuildRoleList(guild --[[optional]])
 end
 
 recent_role_cache = {}
+function FindGuildForRole(roleId)
+	local roleFound
+	local tempGuildList = Config.Guilds
+	tempGuildList["main_guild_abcd1010"] = Config.Guild_ID
+
+  if not roleId then
+    sendDebugMessage("[Badger_Discord_API] ERROR: Invalid usage of FindGuildForRole. Requires `roleId` arg.")
+  end
+  roleId = tostring(roleId)
+	if roleGuilds[roleId] then
+		return roleGuilds[roleId]
+	end
+
+	for _,id in pairs(tempGuildList) do
+		local guild = DiscordRequest("GET", "guilds/"..id, {})
+		if guild.code == 200 then
+			local data = json.decode(guild.data)
+			local roles = data.roles;
+			for i = 1, #roles do
+				if tostring(roles[i].id) == roleId then
+					roleGuilds[roleId] = id
+					roleFound = id
+					break
+				end
+			end
+
+			if roleFound then
+				break
+			end
+		else
+			sendDebugMessage("[Badger_Discord_API] ERROR: please check your config and ensure everything is correct. Error: "..tostring(guild.code))
+		end
+	end
+
+	return roleFound
+end
+
+function RoleNameFromId(id, guild)

```

Comment:
guild should be an optional parameter?

Reply:
It is - looking below the case of guild == nil is handled by defaulting to searching for the guild. Providing the guild so just make it more efficient.